netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -532,7 +532,11 @@ protected void onConnectionError(ChannelHandlerContext ctx, boolean outbound, Th
   @Override
   protected void onStreamError(ChannelHandlerContext ctx, boolean outbound, Throwable cause,
       StreamException http2Ex) {
-    logger.log(Level.WARNING, "Stream Error", cause);
+    Level level = Level.WARNING;
+    if (serverStream == null && http2Ex.error() == Http2Error.STREAM_CLOSED) {
```

Comment:
This won't build. `serverStream` is defined on the next line. The lines will need to be swapped.

Reply:
@ejona86 I have pushed the fix. Please review again.

Reply:
Because the `serverStream` code was moved up to line 535 one unintended consequence is that if that code throws exception (not sure if it ever will) the `logger.log` line will never execute now (which was executing before this change) so we'll not see a log message that used to come out before this change. One way to address it is to put try-catch around line 535. Just pointing it out.

Reply:
That line to get the stream is fairly safe. There may be a few edge cases, but I'm not that worried as there is very little code (transitively) and we'd be seeing NPEs already.